### ISSUE TITLE
Webdav Android client receive error 400

### ISSUE THREAD

**Issue by delumerlino** [owncloud/core#19951]
Actual behaviour

Android is unable to sync with Foldersync and/or Es file explorer using webdav.
Owncloud Android client does not work, but caldav and carddav work
Same situation for OC server 8.1.x, working with OC 8.0.x
Android gets "bad request"
Windows client can connect but receive "error while loading list of folders from the server"

Server configuration

Operating system:
Ubuntu 14.04

Web server:
apache/2.4.16

Database:
Mysql 5.5.44-0ubuntu0.14.04.1
PHP version:
PHP 5.6.14-1+deb.sury.org~trusty+1

ownCloud version: 
ownCloud 8.2 (stable)

Updated from an older ownCloud or fresh install:
Upgraded from previous stable version
moved data directory, saved config.php and placed it into the new owncloud version and set the new data directory. The upgrade worked well, the Windows clients can connect but Android cannot sync anymore.

List of activated apps:
Deleted files  0.6.3
Share Files  0.6.2
Updater  0.6
Versions  1.0.6
Calendar  0.8.0
Contacts  0.5.0.0

The content of config/config.php:

<?php
$CONFIG = array (
  'instanceid' => 'oca5xsros0qm',
  'passwordsalt' => 'delumerlino',
  'secret' => 'delumerlino',
  'trusted_domains' => 
  array (
    0 => 'internal',
  'external',
  'external:port'
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'overwrite.cli.url' => 'https://',
  'dbtype' => 'mysql',
  'version' => '8.2.0.12',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'oc_admin',
  'dbpassword' => '',
  'logtimezone' => 'UTC',
  'installed' => true,
  'theme' => '',
  'mail_smtpmode' => 'smtp',
  'mail_from_address' => 'name',
  'mail_domain' => 'domain.com',
  'mail_smtpauthtype' => 'PLAIN',
  'mail_smtpauth' => true,
  'mail_smtpname' => 'name@domain.com',
  'mail_smtppassword' => 'password',
  'mail_smtphost' => 'smtp.domain.com',
  'mail_smtpport' => 'port',
  'mail_smtpsecure' => 'tls',

  'loglevel' => 0,
  'maintenance' => false,
);

Are you using external storage, if yes which one:
no

Are you using encryption:
no

Client configuration

Operating system:
android 5.0

Logs

ownCloud log (data/owncloud.log)

Exception: {"Message":"HTTP\/1.1 401 No basic authentication headers were found","Exception":"Sabre\\DAV\\Exception\\NotAuthenticated","Code":0,"Trace":"#0 \/var\/www\/owncloud\/lib\/private\/connector\/sabre\/auth.php(148): Sabre\\DAV\\Auth\\Backend\\AbstractBasic->authenticate(Object(OC\\Connector\\Sabre\\Server), 'ownCloud')\n#1 \/var\/www\/owncloud\/lib\/private\/connector\/sabre\/auth.php(121): OC\\Connector\\Sabre\\Auth->auth(Object(OC\\Connector\\Sabre\\Server), 'ownCloud')\n#2 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Auth\/Plugin.php(118): OC\\Connector\\Sabre\\Auth->authenticate(Object(OC\\Connector\\Sabre\\Server), 'ownCloud')\n#3 [internal function]: Sabre\\DAV\\Auth\\Plugin->beforeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#4 \/var\/www\/owncloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#5 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(456): Sabre\\Event\\EventEmitter->emit('beforeMethod', Array)\n#6 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(254): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#7 \/var\/www\/owncloud\/apps\/files\/appinfo\/remote.php(55): Sabre\\DAV\\Server->exec()\n#8 \/var\/www\/owncloud\/remote.php(137): require_once('\/var\/www\/ownclo...')\n#9 {main}","File":"\/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Auth\/Backend\/AbstractBasic.php","Line":70}

Apache log (var/logs/apache2/other_vhosts_access.log)

Owncloud:port x.x.x.x - - [21/Oct/2015:16:53:42 +0200] "PROPFIND /owncloud/remote.php/webdav/ HTTP/1.1" 400 9969 "-" "android-cloud-api - Android 5.0.1"

**Comment by delumerlino** [comment]
...In other words....only web uploading/downloading works....

**Comment by ontheair81** [comment]
Seems to be referenced to https://github.com/owncloud/core/issues/18568

**Comment by delumerlino** [comment]
Solved with allow access in config.php at the host. Now seems that the host declaration is key sensitive.
Anyway, started from scratch with reinstalling all and reimported in the DB the needed tables.

**Comment by ontheair81** [comment]
delumerlino, how did you solve this topic? Please tell the other user affected by the bug your steps to solve it. What do you mean with "host declaration" in the config.php of OwnCloud?

**Comment by ontheair81** [comment]
I can confirm, that the problem is still present after ugrading to 8.2.0. Still no uploads from Android working. I did the upgrade by command line, not via the updater app.

@delumerlino how did you solve it? Please provide more informations.

**Comment by delumerlino** [comment]
Of course!
I wasn't able to fix, so I tried with a fresh install. But after this, it
wasn't so easy to make it work.
I have found that the host name inside config.php (trusted site) must be
the exact name you connect to, it is case sensitive.
I was connecting to my site (fqdn) with first letter capitol, and I have
declared trusted site with no capitol.
But after this, something wasn't working again. I have googled a lot of
pages and tried with this:
You have to accept all the certificates (if you use SSL encryption) in Dav
configuration.  Add these lines at the end of
/owncloud/3dparty/sabre/dav/lib/dav/client.php

CURLOPT_SSL_VERIFYPEER => 0,
CURLOPT_SSL_VERIFYHOST => 0,

Let me know if this is the solution for you! Mine is working great now!
Cheers!
Il 25/Ott/2015 07:23, "ontheair81" <notifications@github.com> ha scritto:

**Comment by ontheair81** [comment]
Thank you for the informations, @delumerlino !

>I have found that the host name inside config.php (trusted site) must be the exact name you connect to, it is case sensitive.

Ok, I double-checked that. Trusted sites are all lower-case, same as I connect on all my devices. No difference.

>You have to accept all the certificates (if you use SSL encryption) in Dav configuration.

I added the lines and it still does not solve the problem. I am using a self-signed certificate. But I did another test and connected from Android without SSL: Still the upload is not working from Android. So in my case its not a problem related to SSL.

Really, really strange. I would suppose, that you solved the problem mostly because of your fresh install. Maybe its the only way to get uploads from Android working again for some odd reason.
